src/terminal-parser.cpp

```diff
@@ -61,10 +61,11 @@ namespace librealsense
             data_vec.insert(data_vec.begin(), data.begin(), data.end());
             return data_vec;
         }
-        else
+        else if (std::all_of(response.begin() + 1, response.end(), [](int x) { return x == 0; })) // if the response contains only the opcode, the response is empty
```

Comment:
vector is of `uin8_t` type, is using `int` here OK?

Reply:
Also as discussed, no need to read all buffer.
User can read a few MB of data and since the vector size if more than 4 it's enough to know we have data to read.
This will loop on all bytes in the buffer